### DISCUSSION TITLE
Solis rai-3k-48es-5g PV figures

### DISCUSSION THREAD

**Discussion by MatcherMike** [wills106/homeassistant-solax-modbus#409] [Q&A]
HI All,

I have the above inverter and have this integration working. I can't see a sensor for the PV generation that i have through CT clamp. I can see the figures on soliscloud. This is correctly reporting my grid meter values.

Is there one?

Thanks
Mike

**Comment by alienatedsec** [comment]
@MatcherMike `PV Total Power` sensor? Are you using plugin_solis_old?

**Reply by MatcherMike** [reply]
Thank you for your reply

This is my sensor.solismodbus_pv_total_power

![pv total power](https://github.com/wills106/homeassistant-solax-modbus/assets/121827283/cbcdc1aa-f820-4fe6-8bc6-eb635d8280ab)

I don't think i'm using plugin_solis_old. How would i tell? I'm using this version 2023.03.1

**Reply by alienatedsec** [reply]
@MatcherMike firstly, it is an old version, please go to HACS >> Integrations >> SolaX.... and re-download the latest version.

Secondly, you can choose which plugin to use when you configure the integration - click on configure and you will be able to amend.

**Reply by MatcherMike** [reply]
Thanks,

I have updated to 2023.05.01. My inverter is set to Solis

![modbus](https://github.com/wills106/homeassistant-solax-modbus/assets/121827283/56215fb0-bca3-4cbe-83d1-5ddec6a70650)

**Reply by MatcherMike** [reply]
I get some values ok. I wondered if PV is for built in PV panels on for example a hybrid, but i have a CT clamp on my PV, and this is set to PV on the inverter.

![pv tool](https://github.com/wills106/homeassistant-solax-modbus/assets/121827283/0b9f2f0e-f960-4bd4-adc7-3a79e5c20faa)
![meter](https://github.com/wills106/homeassistant-solax-modbus/assets/121827283/201aa332-3c72-42f2-8d8b-a0d66fe95dda)

**Reply by alienatedsec** [reply]
@MatcherMike Please, change it to Solis Old

**Reply by MatcherMike** [reply]
Ok done, and rebooted, but now i get no values:

![done](https://github.com/wills106/homeassistant-solax-modbus/assets/121827283/a09b2bb4-6c8d-46d9-86fa-f2a90aabfb73)
![broken 1](https://github.com/wills106/homeassistant-solax-modbus/assets/121827283/a124f7d4-1b67-40da-9f03-0cc15d65a1e6)
![broken 3](https://github.com/wills106/homeassistant-solax-modbus/assets/121827283/c2486da8-adae-41e1-a2ef-3ca8bd601cbf)
a-3886-41da-bdcc-033ee110cd63)
![broken 2](https://github.com/wills106/homeassistant-solax-modbus/assets/121827283/a68df923-5359-44ed-ae47-af76a92455ff)

**Reply by alienatedsec** [reply]
Reload the integration and post any logs if still not working.

**Reply by alienatedsec** [reply]
If only `solis` and not `solis_old` is working for you, it might be that your inverter firmware is old and has different registers to cover the PV generation sensor. I can also see that your PV1 and PV2 are not covered. What is the serial number?

**Reply by MatcherMike** [reply]
Thank you for your help with this.

It seems my serial may not be supported?

My serial starts 0602 and is running software 11000E

![serial](https://github.com/wills106/homeassistant-solax-modbus/assets/121827283/4bca1b6e-2c31-4a21-b9b2-1b341ef0d434)

**Reply by wills106** [reply]
@MatcherMike  Is you Inverter an AC Battery Inverter then? With no built in PV Inputs?

**Reply by MatcherMike** [reply]
Hi, thanks for looking at this, Correct it has no PV inputs. I wondered if this is why it reports 0. I do have a CT clamp from my solis for my PV.  The PV is connected to an unmanaged inverter. I would like this figure.

**Reply by wills106** [reply]
I might need further work on the Serial detection to split out AC & Hybrid models.
Have you checked the disabled by default sensors in case it's shown on one of them?
I can have another look over the docs to see if there is anything obvious.

**Reply by MatcherMike** [reply]
Thanks! Yes i had a look through all the ones showing as disabled by integration but i can't see a figure there. Appreciate your help.
